pkg/chart/chart.go

```diff
@@ -309,35 +312,64 @@ func (c *Chart) InitNetClient(details *Details) (HTTPClient, error) {
 		caCertPool = x509.NewCertPool()
 	}
 
-	// If additionalCA is set, load it
-	customCA := details.Auth.CustomCA
 	namespace := os.Getenv("POD_NAMESPACE")
 	if namespace == "" {
 		namespace = defaultNamespace
 	}
-	if customCA != nil {
-		caCertSecret, err := c.kubeClient.CoreV1().Secrets(namespace).Get(customCA.SecretKeyRef.Name, metav1.GetOptions{})
+
+	// If the app repo url is specified then we'll grab the customCA and
+	// defaultHeaders from there, otherwise default to the current details.Auth
+	// (until that path is removed).
+	// TODO(absoludity): Check whether there is a reason that Details.Auth is a different type
+	// to appRepo.Spec.Auth (though they have identical fields). The following would be simpler
+	// if they were the same types, as we could just find the Auth struct, rather than secret
+	// key refs.
+	var customCASecretRef *corev1.SecretKeySelector
+	var authHeaderSecretRef *corev1.SecretKeySelector
+	defaultHeaders := http.Header{"User-Agent": []string{c.userAgent}}
+	if details.AppRepositoryResourceName != "" {
+		appRepo, err := c.appRepoClient.KubeappsV1alpha1().AppRepositories(namespace).Get(details.AppRepositoryResourceName, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("unable to get app repository %q: %v", details.AppRepositoryResourceName, err)
+		}
+		if appRepo.Spec.Auth.CustomCA != nil {
+			customCASecretRef = &appRepo.Spec.Auth.CustomCA.SecretKeyRef
+		}
+		if appRepo.Spec.Auth.Header != nil {
+			authHeaderSecretRef = &appRepo.Spec.Auth.Header.SecretKeyRef
+		}
+
+	} else {
+		if details.Auth.CustomCA != nil {
```

Comment:
I would leave a comment saying that we are leaving this for backward compatibility and it should be removed.

Reply:
There is a comment to that effect at the start of the block above? ("otherwise default to the current details.Auth (until that path is removed)") Let me know if it's not clear enough.